src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1667,10 +1667,18 @@ private void ReportIfUnused(LocalFunctionSymbol symbol)
 
         public override BoundNode VisitLocal(BoundLocal node)
         {
+            LocalSymbol localSymbol = node.LocalSymbol;
+            if ((localSymbol as SourceLocalSymbol)?.IsVar == true && localSymbol.ForbiddenZone?.Contains(node.Syntax) == true)
```

Comment:
I must have missed a PR... what's `ForbiddenZone`?

Reply:
It is a region of syntax where you're not allowed to reference the variable.  For example inside the initialization of a `var`-declared local, or inside the invocation that declares an `out var` argument.